core/codegen/arithmetic.go

```diff
@@ -0,0 +1,345 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codegen
+
+import (
+	"fmt"
+	"reflect"
+
+	"llvm/bindings/go/llvm"
+)
+
+// Zero returns a zero-value of the specified type.
+func (b *Builder) Zero(ty Type) *Value {
+	if ty == b.m.Types.Void {
+		return nil
+	}
+	return b.val(ty, llvm.ConstNull(ty.llvmTy())).SetName("zero")
+}
+
+// One returns a one-value of the specified bool, int or float type.
+func (b *Builder) One(ty Type) *Value {
+	var value *Value
+	switch {
+	case ty == b.m.Types.Bool:
+		value = b.val(ty, llvm.ConstInt(ty.llvmTy(), 1, false))
+	case IsInteger(ty):
+		value = b.val(ty, llvm.ConstInt(ty.llvmTy(), 1, false))
+	case IsFloat(ty):
+		value = b.val(ty, llvm.ConstFloat(ty.llvmTy(), 1))
+	default:
+		fail("One does not support type %T", ty)
+	}
+	value.SetName("1")
+	return value
+}
+
+// Not returns !x. The type of x must be Bool.
+func (b *Builder) Not(x *Value) *Value {
+	assertTypesEqual(x.ty, b.m.Types.Bool)
+	return b.val(b.m.Types.Bool, b.llvm.CreateNot(x.llvm, ""))
+}
+
+// Not returns ~x.
```

Comment:
I think this was my fault, but this should be "Invert returns ~x"

Reply:
Done